test-interface/src/main/scala/scala/scalanative/testinterface/TestMain.scala

```diff
@@ -31,53 +31,27 @@ object TestMain {
   final val iPv4Loopback = "127.0.0.1"
   final val iPv6Loopback = "::1"
 
-  private def getFreeBSDLoopbackAddr(): String = {
+  private def setFreeBSDWorkaround(): Unit = {
     /* Standard out-of-the-box FreeBSD differs from Linux & macOS in
      * not allowing IPv4 mapped IPv6 addresses, such as :FFFF:127.0.0.1
-     * or ::ffff:7f00:1.  These can be enabled by setting the line
-     * ipv6_ipv4mapping="YES" in /etc/rc.conf (and rebooting).
+     * or ::ffff:7f00:1.  These can be enabled at runtime by running
      *
-     * FreeBSD TestMain initial connections should succeed on both IPv4
-     * and IPv6 systems without requiring arcane and non-standard system
-     * configuration.  This method checks the protocol that Java connect()
-     * is likely used and returns the corresponding loopback address.
-     * Tests which use IPv4 addresses, either through hard-coding or
-     * bind resolution, on IPv6 systems will still fail. This allows to
-     * run the vast majority of tests which do not have this characteristic.
+     *   sysctl net.inet6.ip6.v6only=0
      *
-     * Networking is complex, especially on j-random systems: full of
-     * joy & lamentation.
+     * To make it persistent on reboot it is possible to set the line
+     * ipv6_ipv4mapping="YES" in /etc/rc.conf.
+     *
+     * Another difference is that Java versions >= 11 on FreeBSD set
+     * java.net.preferIPv4Stack=true by default, so the sbt server
+     * listens only on a tcp4 socket.
+     *
+     * The easiest way to make TestMain to work on most FreeBSD machines
+     * with different Java versions without changing system defaults is
+     * to set java.net.preferIPv4Stack=true in Scala Native in order to
```

Comment:
s/to set/ to set, __before__ the first Java network call,/

or such. In SN, the check for preferIPv4Stack is done once,
before the first Java network call.